spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/DataFlowTemplate.java

```diff
@@ -163,8 +163,13 @@ public DataFlowTemplate(URI baseURI, RestTemplate restTemplate) {
 		final RootResource resourceSupport = restTemplate.getForObject(baseURI, RootResource.class);
 
 		if (resourceSupport != null) {
-			String serverRevision = resourceSupport.getApiRevision() != null
-					? resourceSupport.getApiRevision().toString() : "[unknown]";
+			if (resourceSupport.getApiRevision() == null) {
+				throw new IllegalStateException("Incompatible version of Data Flow server detected.\n"
```

Comment:
Given server incompatibility is handled separately [here](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/DataFlowTemplate.java#L171), do you think we can have even more generic documentation link at this line?